### PR TITLE
Add `email_recipient_filter` column to posts table

### PR DESCRIPTION
**Pull request by allouis**
no-issue

This will be used to store the intended recipients for a newsletter email, and will allow us to decouple newsletter recipients from the visibility of a post.

There are two migrations here:
1. Add the new column
2. Populate the new column based on the existing visibility logic

### CODE REVIEW

**Review by rshbhgrg — APPROVED** [review]

**Review by daniellockyer — APPROVED** [review]

File: core/server/data/schema/schema.js [hunk 1]
```diff
@@ -453,6 +453,13 @@ module.exports = {
             defaultTo: 'pending',
             validations: {isIn: [['pending', 'submitting', 'submitted', 'failed']]}
         },
+        recipient_filter: {
+            type: 'string',
+            maxlength: 50,
+            nullable: false,
+            defaultTo: 'paid',
```
Comment (allouis) [inline core/server/data/schema/schema.js:460]: What do you think about this default value? It felt the safest for me, and we need a default value when adding a new column if the column is not nullable (AFAIK?)

File: core/server/data/migrations/versions/3.38/04-populate-recipient-filter-column.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        await connection('emails')
+            .join('posts', 'emails.post_id', 'posts.id')
+            .update('emails.recipient_filter', 'all')
+            .where('posts.visibility', 'public');
+    },
+
+    async function down(connection) {
+        logging.info('Updating emails.recipient_filter values to paid');
+        await connection('posts')
+            .update('email_recipient_filter', 'paid');
```
Comment (daniellockyer) [inline core/server/data/migrations/versions/3.38/04-populate-recipient-filter-column.js:27]: Code + comment does different things. Should be `await connection('emails').update('recipient_filter', 'paid');`

**Review by daniellockyer — CHANGES_REQUESTED** [review]

File: core/server/data/migrations/versions/3.38/04-populate-recipient-filter-column.js (hunk 2 above)
Comment (allouis) [reply core/server/data/migrations/versions/3.38/04-populate-recipient-filter-column.js]: Fixed!

**Review by kevinansfield — APPROVED** [review]

**Review by daniellockyer — APPROVED** [review]

File: core/server/data/schema/schema.js (hunk 1 above)
Comment (allouis) [reply core/server/data/schema/schema.js:460]: We are going to run another migration to remove the default value

**Pull request merged**